internal/provider/resource.go

```diff
@@ -5,47 +5,64 @@ import (
 	"fmt"
 	"math/rand"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func nullResource() *schema.Resource {
-	return &schema.Resource{
-		Description: `The ` + "`null_resource`" + ` resource implements the standard resource lifecycle but takes no further action.
+var (
+	_ resource.Resource = (*nullResource)(nil)
+)
 
-The ` + "`triggers`" + ` argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced.`,
+func NewNullResource() resource.Resource {
+	return &nullResource{}
+}
+
+type nullResource struct{}
+
+func (n nullResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_resource"
+}
 
-		CreateContext: resourceCreate,
-		ReadContext:   resourceRead,
-		DeleteContext: resourceDelete,
+func (n nullResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `The ` + "`null_resource`" + ` resource implements the standard resource lifecycle but takes no further action.
 
-		Schema: map[string]*schema.Schema{
+The ` + "`triggers`" + ` argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced.`,
+		Attributes: map[string]tfsdk.Attribute{
 			"triggers": {
 				Description: "A map of arbitrary strings that, when changed, will force the null resource to be replaced, re-running any associated provisioners.",
-				Type:        schema.TypeMap,
+				Type:        types.MapType{ElemType: types.StringType},
 				Optional:    true,
-				ForceNew:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
```

Comment:
We _may_ want to consider something like https://github.com/hashicorp/terraform-provider-random/pull/305 for this resource as well, but we can discuss and potentially handle this detail before any release.